File: Chapter 03/Python/section_3_3_3_lsp/after/midfielder.py
from .player import Player

"""
LSP SOLUTION: True Substitutability.
* ARCHITECTURE NOTE: Just like the Forward, the Midfielder is 100% 
compatible with the Player abstraction. We can swap one for the other 
at runtime without the Coach ever needing to know the difference.
"""
class Midfielder(Player):
    def play_field_position(self):
        print("  [Midfielder] Controlling the midfield, passing and tackling.")

File: Chapter 03/JS/section_3_3_3_lsp/after/midfielder.js
const Player = require('./player');

/**
 * LSP SOLUTION: True Substitutability.
 * * ARCHITECTURE NOTE: Just like the Forward, the Midfielder is 100% 
 * compatible with the Player abstraction. We can swap one for the other 
 * at runtime without the Coach ever needing to know the difference.
 */
class Midfielder extends Player {
    playFieldPosition() {
        console.log("  [Midfielder] Controlling the midfield, passing and tackling.");
    }
}

module.exports = Midfielder;